google-cloud-firestore/lib/google/cloud/firestore/document_snapshot.rb

```diff
@@ -285,11 +308,47 @@ def missing?
           grpc.nil?
         end
 
+        ##
+        # @private
+        def <=> other
+          return nil unless other.is_a? DocumentSnapshot
+          grpc <=> other.grpc
+        end
+
+        ##
+        # @private
+        def eql? other
+          return nil unless other.is_a? DocumentSnapshot
+          grpc.eql? other.grpc
```

Comment:
Is `grpc` the backing Protobuf? We have decided to not include the metadata (read time, update time, create time) in the equals function since it makes the API practically unusable. No Documents would be the same unless their pointer addresses are the same.

Feel free to address in a follow up if you think this is an actual issue.

Reply:
Yes, `grpc` is the protobuf object. 

This will affect not only equals, but the hash value as well. I’ll adjust these methods to ignore these date values.

Should empty snapshots be ordered before or after snapshots with data? Assuming the resource names are the same?

Reply:
That should never happen (you should never have the same document with two different sets of data in the same data structure). If you do want to address this, then feel free to pick an ordering that suits your liking.

Reply:
These methods are private and here to make things like lookup in a Hash more efficient. They are not used by the Watch ordering. I've updated them to not consider the date values.